server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -638,7 +649,12 @@ private Tuple<RemoteStoreRefreshListener, RemoteStoreStatsTrackerFactory> mockIn
         RemoteStoreSettings remoteStoreSettings = mock(RemoteStoreSettings.class);
         when(remoteStoreSettings.getMinRemoteSegmentMetadataFiles()).thenReturn(10);
         when(shard.getRemoteStoreSettings()).thenReturn(remoteStoreSettings);
-        RemoteStoreRefreshListener refreshListener = new RemoteStoreRefreshListener(shard, emptyCheckpointPublisher, tracker);
+        RemoteStoreRefreshListener refreshListener = new RemoteStoreRefreshListener(
+            shard,
+            emptyCheckpointPublisher,
+            tracker,
+            DefaultRemoteStoreSettings.INSTANCE
+        );
         refreshListener.afterRefresh(true);
```

Comment:
Can we have a test for the changes where we are hitting the time-limit?

Reply:
yes, working on adding tests